src/app/components/playlist_details/playlist_details_model.rs

```diff
@@ -109,6 +130,34 @@ impl PlaylistModel for PlaylistDetailsModel {
         self.app_model.get_state().playback.current_song_id()
     }
 
+    fn playlist_song_ids(&self) -> Option<Vec<String>> {
+        if let Some(playlist) = self.get_playlist_info() {
+            let playlist_ids = playlist
+                .songs
+                .songs
+                .iter()
+                .map(|song_d| song_d.id.clone())
+                .collect::<Vec<_>>();
+            return Some(playlist_ids);
+        }
+        None
+    }
+
+    fn playlist_is_playing(&self) -> bool {
```

Comment:
there is a simpler way to implement this logic : you can look in the `PlaybackState` for `source`, you might have to add an accessor there but thats fine :)